version: '2.1'

services:
  mai-db-node-ex01:
    build:
      context: mariadb
      dockerfile: Dockerfile
    container_name: mai-db-node-ex01
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: messenger
      MYSQL_USER: admin
      MYSQL_PASSWORD: password1
      MYSQL_ROOT_PASSWORD: '1'
    command: >
      --sql-mode='STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'
      --general-log=ON
      --log-queries-not-using-indexes=ON
    networks:
      - arch-network
    ports:
      - 3360:3306
    healthcheck:
      test: "/usr/bin/mysql --user=admin --password=password1 --execute \"SHOW DATABASES;\""
      timeout: 20s
      retries: 5
    volumes:
      - db-node-ex01-data:/var/lib/mysql

  web_server:
    build:
      context: web_server
      dockerfile: Dockerfile
    container_name: web_server
    environment:
      - DB_HOST=mai-db-node-ex01
      - DB_PORT=3306
      - DB_LOGIN=admin
      - DB_PASSWORD=password1
      - DB_DATABASE=messenger
    ports:
      - 8080:8080
    networks:
      - arch-network
    depends_on:
      mai-db-node-ex01:
        condition: service_healthy
  other_server:
    build:
      context: other_server
      dockerfile: Dockerfile
    container_name: other_server
    environment:
      - SERVICE_HOST=web_server
    ports:
      - 8081:8081
    networks:
      - arch-network

volumes:
  db-node-ex01-data:

networks:
  arch-network:
    driver: "bridge"
    external: false
